### PR TITLE
[feature](Paimon) support deletion vector for Paimon naive reader

### PR DESCRIPTION
**Pull request by suxiaogang223**
## Proposed changes

support deletion vector for Paimon naive reader

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/format/orc/vorc_reader.cpp [hunk 1]
```diff
@@ -1750,6 +1750,11 @@ Status OrcReader::get_next_block(Block* block, size_t* read_rows, bool* eof) {
                 RETURN_IF_CATCH_EXCEPTION(Block::filter_block_internal(block, columns_to_filter,
                                                                        (*_delete_rows_filter_ptr)));
             }
+            // TODO: Check compatibility of deletion vector and lazy reading
+            IColumn::Filter result_filter(block->rows(), 1);
+            _execute_filter_position_delete_rowids(result_filter);
```
Comment (AshinGau) [inline be/src/vec/exec/format/orc/vorc_reader.cpp:1755]: Is there a performance issue with a regular hive table？

Comment (suxiaogang223) [reply be/src/vec/exec/format/orc/vorc_reader.cpp:1755]: These changes also happened in https://github.com/apache/doris/pull/34814. I will rebase when this pr merged.

**Comment by suxiaogang223** [comment]
run buildall

**Review by AshinGau — APPROVED** [review]

**Comment by AshinGau** [comment]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Comment by morningman** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by AshinGau — APPROVED** [review]

**Pull request merged**